### PR TITLE
Add CI tests for conda-standalone canaries

### PR DESCRIPTION
**Pull request by jaimergp**
### Description

We've been doing some work over at `conda/conda-standalone` so it can be maintained like a proper open source project. This means we have now canaries we can use to test new versions with constructor.

### Checklist - did you ...

- [x] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [x] Add / update necessary tests?
- [ ] Add / update outdated documentation?

### CODE REVIEW

**Comment by jaimergp** [comment]
Everything seems to work. Let's add it to the CI.

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -31,28 +32,55 @@ defaults:
...
-            micromamba: true
+            conda-standalone: conda-standalone
             check-docs: true
+          - os: ubuntu
+            python-version: "3.10"
+            conda-standalone: conda-standalone-nightly
+          - os: ubuntu
+            python-version: "3.11"
+            conda-standalone: micromamba
+          # MACOS
```
Comment (kenodegard) [inline .github/workflows/main.yml:56]: are you intentionally skipping Python 3.8 for macOS?

**Review by kenodegard — DISMISSED** [review]

File: .github/workflows/main.yml (hunk 1 above)
Comment (jaimergp) [reply .github/workflows/main.yml:56]: Accidental at first, then I thought, hm why not? macOS runners are scarce after all. Then I asked in the call whether we can drop or not for `conda/conda`. I guess I'll add it back?

Comment (kenodegard) [reply .github/workflows/main.yml:56]: In conda-build we opted to only test the lower and upper versions on macOS. My thought behind that was we'd catch stdlib deprecations with the upper version and we'd catch usage of newer stdlib features that the lower version doesn't support.

Dunno how relevant that would be for `constructor`, just wanted to be sure the exclusion was intentional. 👍🏻

Comment (jaimergp) [reply .github/workflows/main.yml:56]: I like that. Will apply changes and ping you again. Thanks!

Comment (jaimergp) [reply .github/workflows/main.yml:56]: Done @kenodegard. Can you reapprove? Thanks 🙏

**Review by kenodegard — APPROVED** [review]

**Pull request merged**